tests/template_tests/test_logging.py

```diff
@@ -69,3 +69,38 @@ def test_log_on_variable_does_not_exist_not_silent(self):
     def test_no_log_when_variable_exists(self):
         Variable('article.section').resolve({'article': {'section': 'News'}})
         self.assertIsNone(self.test_handler.log_record)
+
+
+class ExceptionDuringRenderLoggingTests(SimpleTestCase):
+    def setUp(self):
+        self.test_handler = TestHandler()
+        self.logger = logging.getLogger('django.template')
+        self.original_level = self.logger.level
+        self.logger.addHandler(self.test_handler)
+        self.logger.setLevel(logging.WARN)
+
+    def tearDown(self):
+        self.logger.removeHandler(self.test_handler)
+        self.logger.level = self.original_level
+
+    def test_logs_exceptions_during_rendering_with_debug_disabled(self):
+        engine = Engine(loaders=[
+            ('django.template.loaders.locmem.Loader', {
+                'child': '{% for var in varlist %}{{ var.somemethod }}{% endfor %}',
+            }),
+        ], debug=False)
+        request = RequestFactory().get('/')
+
+        class ErrorObj():
+            def somemethod(self):
+                raise IndexError("some generic exception")
+
+        always_raises_exception = {'varlist': [ErrorObj()]}
```

Comment:
I don't believe this is sufficient but will double check as I'm simplifying other aspects of the test. I believe a method call is necessary otherwise the raised exception is handled via another code path.